paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/BatchWriteGeneratorTagOperator.java

```diff
@@ -114,15 +117,15 @@ private void createTag() {
         try {
             // If the tag already exists, delete the tag
             if (tagManager.tagExists(tagName)) {
-                tagManager.deleteTag(tagName, tagDeletion, snapshotManager);
+                tagManager.deleteTag(tagName, tagDeletion, snapshotManager, callbacks);
```

Comment:
Be careful here, you need to create new callbacks, the callbacks will be closed here.

Reply:
You are right, but I see some other same usage eg: `TagAutoCreation#callbacks`?  I think it's the same error, do i have to fix it in this PR ?

Reply:
Yes, you can fix them together.